File: MoonEngine-Root/ECS/src/main/java/com/greentree/engine/moon/ecs/ComponentLock.java
package com.greentree.engine.moon.ecs;

import com.greentree.engine.moon.ecs.component.Component;

public interface ComponentLock extends AutoCloseable {
	
	@Override
	void close();
	
	void remove(Class<? extends Component> componentClass);
	
	void add(Component component);
	
	<C extends Component> C get(Class<C> componentClass);
	
}

File: MoonEngine-Root/ECS/src/main/java/com/greentree/engine/moon/ecs/ComponentLock.kt
package com.greentree.engine.moon.ecs

import com.greentree.engine.moon.ecs.component.Component

interface ComponentLock : AutoCloseable {

	override fun close()
	fun remove(componentClass: Class<out Component>)
	fun add(component: Component)
	operator fun <C : Component> get(componentClass: Class<C>): C
}